content/articles/2019-09-tidyr-1-0-0.Rmarkdown

```diff
@@ -0,0 +1,316 @@
+  can be broken down into two simpler steps, called `pack()` and `chop()`.
+  
+* New `expand_grid()`, a variant of `base::expand.grid()`. This is a useful 
+  function to know about, but also serves as a good reason to talk about the 
+  important role that [vctrs](http://vctrs.r-lib.org/) plays behind the scenes. 
+  You shouldn't ever _have_ to leave about vctrs, but it brings improvements to 
+  consistency and performance. 
+
+As well as implementing the new features, I've spent considerable time on the documentation, including four major new vignettes:
+
+* [`vignette("pivot")`](https://tidyr.tidyverse.org/dev/articles/pivot.html), 
+  [`vignette("rectangle")`](https://tidyr.tidyverse.org/dev/articles/rectangle.html), 
+  and [`vignette("nest")`](https://tidyr.tidyverse.org/dev/articles/nest.html) 
+  provide detailed documentation and case studies of pivotting, rectangling, 
+  and nesting respectively.
+
+* [`vignette("in-packages")`](https://tidyr.tidyverse.org/dev/articles/in-packages.html)
+  provides best practices for using tidyr inside
+  another package, and provides detailed advice on working with multiple 
+  versions of tidyr if an interface change has affected your package.
+
+You can see a complete list of all the other minor bug fixes and improvements in the [release notes](https://tidyr.tidyverse.org/dev/news/index.html#tidyr-1-0-0).
+
+```{r setup, message = FALSE}
+library(tidyr)
+library(dplyr)
+```
+
+## Pivoting
+
+New [`pivot_longer()`](https://tidyr.tidyverse.org/dev/reference/pivot_longer.html) and [`pivot_wider()`](https://tidyr.tidyverse.org/dev/reference/pivot_wider.html) provide modern alternatives to `spread()` and `gather()`. They have been carefully redesigned to be easier to learn and remember, and include many new features. `spread()` and `gather()` won't go away, but they've been retired which means that they're no longer under active development.
+
+The best place to learn about `pivot_longer()` and `pivot_wider()` is [`vignette("pivot")`](https://tidyr.tidyverse.org/dev/articles/pivot.html), or by watching my presentation to the [Vienna R users group](https://www.youtube.com/watch?v=D48JHU4llkk). Here I'll quickly show off a few of the coolest new features:
+
+*   `pivot_longer()` can now separate column names into multiple variables in 
+    a single step. For example, take the `who` dataset which has column names
+    that look like `new_{diagonsis}_{gender}{age}`:
+    
+    ```{r}
+    names(who)
+    ```
+    
+    You can now tease apart the variable names in a single step by supplying
+    a vector of variable names to `names_to` and a regular expression to
+    `names_pattern` (simpler cases might only need `names_sep`):
+
+    ```{r}
+    who %>% pivot_longer(
+      cols = new_sp_m014:newrel_f65,
+      names_to = c("diagnosis", "gender", "age"), 
+      names_pattern = "new_?(.*)_(.)(.*)",
+      values_to = "count"
+    )
+    ```
```

Comment:
It's a bummer that all the visible data is `NA` for `count`. Can you work with a different dataset or a different slice?

Reply:
I'll just add `values_drop_na = TRUE`